File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n - len(lost)
    lost.sort()
    reserve.sort()
    
    reserveDict = {}
    for r in reserve:
        if r not in lost:
            reserveDict[r] = 1
        else:
            answer+=1
    for l in lost:
        if l in reserve:
            continue
        if l - 1 in reserveDict and reserveDict[l-1] > 0:
            reserveDict[l-1] -=1
            answer+=1
        elif l + 1 in reserveDict and reserveDict[l+1] > 0:
            reserveDict[l+1] -=1
            answer+=1
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] dp = new int[n + 1];
        for(int i = 1; i<=n;i++){
            dp[i] = 1;
        }
        for(int i =0; i<lost.length; i++){
            dp[lost[i]]--;
        }
        for(int i =0; i<reserve.length; i++){
            dp[reserve[i]]++;
        }
        
        for(int i = 1; i<= n;i++){
            if(dp[i] == 0){
                if(i-1>=1 && dp[i-1]==2){
                    dp[i] = 1;
                    dp[i-1]--;
                }
                else if (i+1<=n && dp[i+1]==2){
                    dp[i] = 1;
                    dp[i+1]--;
                }
            }
        }
        
        int answer = 0;
        for(int i = 1; i<= n;i++){
            if(dp[i] > 0){
                answer++;
            }
        }
        return answer;
    }
}